### PR TITLE
KWS-2821:  Login View skin

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/KWS-2821  

## Description
Added skin functionality to the Login view

## Screenshots
n/a

### CODE REVIEW

File: PopJamUI/Classes/Skins/Models/Components/UITextFieldViewSkinModel.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public var textFieldTextFont: UIFont
+    public var textFieldTintColor: UIColor
+    public var autocorrectionType: UITextAutocorrectionType
+    public var keyboardType: UIKeyboardType
+    public var autocapitalizationType: UITextAutocapitalizationType
+    public var returnKeyType: UIReturnKeyType
+    public var clearButtonMode: UITextFieldViewMode
+    public var contentVerticalAlignment: UIControlContentVerticalAlignment
+    
+    init(placeholderText: String,
```
Comment (me2k) [inline PopJamUI/Classes/Skins/Models/Components/UITextFieldViewSkinModel.swift]: Make Public?

File: PopJamUI/Classes/Skins/Models/Components/UIButtonSkinModel.swift [hunk 2]
```diff
@@ -2,29 +2,55 @@
...
-                selectedColor: UIColor,
-                iconImageName: String?  =  "",
-                title: String?  =  "",
-                buttonStyle: KWSButtonType) {
-        self.mainColor = mainColor
-        self.selectedColor = selectedColor
-        self.iconImageName = iconImageName
-        self.title = title
-        self.buttonStyle = buttonStyle
+    init(normalImageName: String? = "",
```
Comment (me2k) [inline PopJamUI/Classes/Skins/Models/Components/UIButtonSkinModel.swift]: Make Public?

File: PopJamUI/Classes/Utilis/Themes.swift [hunk 3]
```diff
@@ -15,17 +15,84 @@ public struct Themes {
...
+                                                              loginButton: KWSButtonSkinModel.init(mainColor: UIColor().PJBlue(),
+                                                                                                   selectedColor: UIColor().PJBlue_Pressed(),
+                                                                                                   iconImageName: "",
+                                                                                                   title: Bundle_Resource_Loader.loadLocalizedText(key: "InitialViewLogInButtonTitleKey")!,
+                                                                                                   buttonStyle: .fullColor),
+                                                              viewBackgroundColor: .white)
+    
+    public let loginTheme = LogInViewSkinModel.init(topBarImage: UIImageSkinModel.init(image: Bundle_Resource_Loader.loadImage(name: "topLine_bar")!),
+                                                    userNameTextField: KWSTextFieldViewSkinModel.init(type: .userName,
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/Utilis/Themes.swift]: Should be renamed into `TextFieldViewSkinModel`

File: PopJamUI/Classes/Utilis/Themes.swift [hunk 4]
```diff
@@ -15,17 +15,84 @@ public struct Themes {
...
-                                                                                                     iconImageName: "",
-                                                                                                     title: "CREATE ACCOUNT",
-                                                                                                     buttonStyle: .borderColor),
-                                                            loginButton: UIButtonSkinModel.init(mainColor: UIColor().PJBlue(),
-                                                                                                selectedColor: UIColor().PJBlue_Pressed(),
-                                                                                                iconImageName: "",
-                                                                                                title: "SIGN UP",
-                                                                                                buttonStyle: .fullColor),
+                                                            getStartedButton: KWSButtonSkinModel.init(mainColor: UIColor().PJBlue(),
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/Utilis/Themes.swift]: Should be `ButtonSkinModel`

File: PopJamUI/Classes/Skins/Protocols/Components/KWSTextFieldViewSkinProtocol.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+//  PopJamUI
+//
+//  Created by Tiziano Bruni on 18/05/2018.
+//
+
+import Foundation
+import UIKit
+
+// MARK: KWSTextFieldView Skin
+public protocol KWSTextFieldViewSkinProtocol {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/Skins/Protocols/Components/KWSTextFieldViewSkinProtocol.swift]: Not ObjC, don't need the ugly KWS prefix.

File: PopJamUI/Classes/Skins/Protocols/Components/KWSButtonSkinProtocol.swift [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+//  PopJamUI
+//
+//  Created by Tiziano Bruni on 18/05/2018.
+//
+
+import Foundation
+import UIKit
+
+// MARK: UIButton Skin
+public protocol KWSButtonSkinProtocol {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/Skins/Protocols/Components/KWSButtonSkinProtocol.swift]: Same here.

File: PopJamUI/Classes/LoginView/ViewModel/LoginViewModel.swift [hunk 7]
```diff
@@ -29,23 +29,23 @@ public class LoginViewModel: LoginViewDataProviderDelegate {
     // MARK: Localised text
     
     public func localisedLoginViewTitle() -> String {
-        return Bundle_Resource_Loader.loadLocalizedText(key: "PJLoginViewTitleKey")!
+        return Bundle_Resource_Loader.loadLocalizedText(key: "LoginViewTitleKey")!
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/LoginView/ViewModel/LoginViewModel.swift]: Shouldn't this be part of the skin? All the texts I mean.

File: PopJamUI/Classes/Utilis/Themes.swift (hunk 3 above)
Comment (Tbruni85) [reply PopJamUI/Classes/Utilis/Themes.swift]: the name is miss leading, KWSTextField is the entire component, containing the UITextField

File: PopJamUI/Classes/InitialView/Views/InitialView.swift [hunk 8]
```diff
@@ -69,10 +73,15 @@ public class InitialView: UIView, UITextViewDelegate {
     func buildTagLineLabel() -> UILabel {
         let label = UILabel.init(frame: .zero)
         label.applySkin(skin: theme.tagLine)
-        label.numberOfLines = 0
         return label
     }
     
+    func buildEmojiImageView() -> UIImageView {
+        let emojiView = UIImageView.init(image: theme.rightSideImage?.image)
+        //emojiView.alpha = Rules.isDemo ? 0.0 : 1.0
```
Comment (me2k) [inline PopJamUI/Classes/InitialView/Views/InitialView.swift]: we should sort this out... my mess.

File: PopJamUI/Classes/LoginView/ViewModel/LoginViewModel.swift (hunk 7 above)
Comment (me2k) [reply PopJamUI/Classes/LoginView/ViewModel/LoginViewModel.swift]: second this

Comment (Tbruni85) [reply PopJamUI/Classes/LoginView/ViewModel/LoginViewModel.swift]: I'm cleaning the ViewModel/DataProvider

File: PopJamUI/Classes/GenericViews/KWSButton.swift [hunk 9]
```diff
@@ -74,7 +74,7 @@ public class KWSButton: UIButton {
         self.setupButton()
     }
     
-    convenience public init(WithSkin skin: UIButtonSkinProtocol) {
+    convenience public init(WithSkin skin: KWSButtonSkinProtocol) {
         self.init(withColor: skin.mainColor,
                   selectedColor: skin.selectedColor,
                   iconImageName: skin.iconImageName!,
```
Comment (me2k) [inline PopJamUI/Classes/GenericViews/KWSButton.swift]: Can you fix this, passing in nil causes the app to crash

**Pull request merged**